core/src/main/java/org/apache/calcite/sql/fun/SqlCaseOperator.java

```diff
@@ -216,9 +219,7 @@ public RelDataType inferReturnType(
       SqlOperatorBinding opBinding) {
     // REVIEW jvs 4-June-2005:  can't these be unified?
     if (!(opBinding instanceof SqlCallBinding)) {
-      return inferTypeFromOperands(
```

Comment:
Just as the comment addressed, can we unify the call bindings ?

Reply:
I ever noticed the comment left by Julian and made some efforts to unify them.
But it seems to be diffcult, especially for the usage of unique fields in `SqlCallBinding`.
Maybe we need a thorough consideration on unifying type inference in different phases (validating/converting/runtime).

Reply:
Thanks for the clarification, i have no more comments.